packages/material-ui/src/withMobileDialog/withMobileDialog.js

```diff
@@ -8,6 +8,11 @@ import withWidth, { isWidthDown } from '../withWidth';
  * Notice that this Higher-order Component is incompatible with server-side rendering.
  */
 const withMobileDialog = (options = {}) => (Component) => {
+  if (process.env.NODE_ENV !== 'production') {
+    console.warn(
+      'Material-UI: The `withMobileDialog` function is deprecated. See https://github.com/mui-org/material-ui/pull/23202.',
```

Comment:
What is this link supposed to do? Linking to the docs (with a permalink) is more robust and digestible for library users since not everybody is using GitHub.

Reply:
If we link the documentation, we will have to update the URL from https://next.material-ui.com/ to https://material-ui.com/ if we need to update the migration instructions. With GitHub, developers get more context, we can update the description if it's not clear enough, they can more easily raise questions about the migration.

I think that both have pros & cons. It would interesting to settle on one approach as we have more deprecations coming. I guess GitHub is the cheap approach, docs could be better.

Reply:
That's why we have the url shortener/redirect route `/r/`. We can easily switch when releasing. 

> With GitHub, developers get more context,

That's not a good thing. The migration action should be focused on what to do not why. This could always be achieved with other means. If that is an argument then why have the migration guide if it should just be a collection of links to PR.